### PR TITLE
Fix truncated output in `fig run` - resolves #658

### LINKED ISSUE #658: Truncated output with `fig run`
Whenever I do a `fig run` command that returns many lines of output, the output is either truncated, or weirdly garbled. This typically happens when running `fig run web rake routes`. The output will typically come back missing 3-5 lines of output (randomly). `docker run -it --rm $IMAGE rake routes` works just fine.

Using fig 1.0.1.

### PR DESCRIPTION
**Pull request by Malet**
Signed-off-by: Michael Malet <MichaelMa@square-enix.com>

### CODE REVIEW

**Comment by Malet** [comment]
rebased again

**Comment by aanand** [comment]
ping @bfirsh @dnephin

File: fig/service.py [hunk 1]
```diff
@@ -300,7 +300,7 @@ def start_container_if_stopped(self, container, **options):
             log.info("Starting %s..." % container.name)
             return self.start_container(container, **options)
 
-    def start_container(self, container, intermediate_container=None, **override_options):
+    def get_start_options(self, container=None, intermediate_container=None, **override_options):
```
Comment (dnephin) [inline fig/service.py]: container should be required, not a kwarg

File: fig/service.py [hunk 2]
```diff
@@ -329,7 +329,14 @@ def start_container(self, container, intermediate_container=None, **override_opt
             dns_search=dns_search,
             restart_policy=restart,
             cap_add=cap_add,
-            cap_drop=cap_drop,
+            cap_drop=cap_drop
+        )
+
+    def start_container(self, container=None, intermediate_container=None, **override_options):
+        container = container or self.create_container(**override_options)
```
Comment (dnephin) [inline fig/service.py]: This reverts an earlier change. `container` should be a required arg, not a kwarg, and this line should be removed.

**Comment by dnephin** [comment]
I think the rebase reverted the change to the `container` arg. Once that's fixed you should squash your commits to a single commit.

**Comment by aanand** [comment]
In addition to fixing the `container` arg regression that @dnephin mentions, this is going to need another rebase as the `fig` directory has now been renamed to `compose`.

**Comment by Malet** [comment]
@dnephin, @aanand - sorry for the delay, how does this look?

**Comment by adrienbrault** [comment]
Tried with that patch, and still having the truncating issue. Not sure if I'm missing something

**Comment by aanand** [comment]
@adrienbrault There are two truncation issues: 

1. Some but not all of a container's output appears in `fig run` - this is a dockerpty bug.
2. A container's output is completely missing from `fig up` - this PR should fix it.

Which one are you seeing?

File: compose/service.py [hunk 3]
```diff
@@ -333,7 +333,14 @@ def start_container(self, container, intermediate_container=None, **override_opt
             dns_search=dns_search,
             restart_policy=restart,
             cap_add=cap_add,
-            cap_drop=cap_drop,
+            cap_drop=cap_drop
+        )
+
+    def start_container(self, container, intermediate_container=None, **override_options):
+        container = container or self.create_container(**override_options)
```
Comment (aanand) [inline compose/service.py]: This line still needs removing. We should always expect `container` to not be None.

**Comment by adrienbrault** [comment]
@aanand I'm seeing 1. Is there an issue tracking this one ?

**Comment by aanand** [comment]
@adrienbrault https://github.com/d11wtq/dockerpty/issues/26

**Comment by dnephin** [comment]
I believe this is fixed now in the 1.3.0 rc

**Pull request closed without merging**